doc/Compilation.md

```diff
@@ -178,6 +178,39 @@ sections are placed in the expected order, adding a section that lists
 necessary load-time relocations, and creating the `LoadInfo` header.
 
 
+### Tock Application Bundle
+
+To support ease-of-use and distributable applications, Tock applications are
+compiled for multiple architectures and bundled together into a "Tock
+Application Bundle" or `.tab` file. This creates a standalone file for an
+application that can be flashed onto any board that supports Tock, and removes
+the need for the board to be specified when the application is compiled.
+The TAB has enough information to be flashed on many or all Tock compatible
+boards, and the correct binary is chosen when the application is flashed
+and not when it is compiled.
+
+#### TAB Format
+
+`.tab` files are `tar`ed archives of TBF compatible binaries along with a
+`metadata.toml` file that includes some extra information about the application.
+A simplified example command that creates a `.tab` file is:
+
+    tar cf app.tab cortex-m0.bin cortex-m4.bin metadata.toml
+
+#### Metadata
+
+The `metadata` file in the `.tab` file is a TOML file that contains a series of
+key-value pairs, one per line, that provides more detailed information and can
+help when flashing the application. Existing fields:
+
+```
+tab-version = 1                         // Version number for this TAB
+name = "<package name>"                 // Package name of the application
+compatible-with = <list of boards>      // List of board kernels that this application supports
```

Comment:
Maybe we want some way to express which driver nums must exist for an app to run at some point, but that is not the goal of this. This is a tool for the programmer to say which boards this app makes sense on. For example, it doesn't make sense to flash the `signpost_audio` app on to the signpost controller, even if all of the required drivers are present.

Right now if you try to violate this, `tockloader` will fail. You can add `--force` to, well, force the app on.

I don't like the idea of '*' or something by default because that adds information without the programmer actually signing off on it. I would rather see the name changed to something that implies that this is a hint, and if nothing is listed you can't derive any information.